### ISSUE TITLE
Allow using a variable name twice in a header

### ISSUE THREAD

**Issue by kawu** [texttheater/produce#50]
First of all, thanks for a great tool!

I've run into an issue with the following specification, where variable `lang` is used twice in a section's header:
```
[]
default = data/fastText/fr/cc.fr.300.bin.gz

[data/fastText/%{lang}/cc.%{lang}.300.bin.gz]
recipe =
  mkdir -p $(dirname %{target})
  wget https://dl.fbaipublicfiles.com/fasttext/vectors-crawl/cc.%{lang}.300.bin.gz -P $(dirname %{target})
```
Running `produce` gives the following error:
```
re.error: redefinition of group name 'lang' as group 2; was group 1 at position 35
```

While there are some easy workarounds (rename the file after download, or just use two variable names), it would be nice if `produce` could support such a use case.

**Comment by texttheater** [comment]
Thanks a lot for the suggestion! This actually duplicates a long-standing TODO in the code, now finally fixed and available in the latest release of Produce (v0.4.1).

**Comment by kawu** [comment]
Great, thanks for the fix!
